Attribute für FoDaKo-DFG hinzugefügt

Zack-83 opened the pull request:
Attribute für die FoDaKo DFG-Kataloge hinzugefügt.


|Attribut                                                           |Änderung       |
|-------------------------------------------------------------------|---------------|
|`deletion                                                         `|neu hinzugefügt|
|`deletion/reasons                                                 `|neu hinzugefügt|
|`deletion/technique                                               `|neu hinzugefügt|
|`sensitive_data/personal_data/consent/content                     `|neu hinzugefügt|
|`sensitive_data/personal_data/consent/how_obtained                `|neu hinzugefügt|
|`sensitive_data/personal_data/protection_concept                  `|neu hinzugefügt|
|`sensitive_data/personal_data/protection_concept/checked          `|neu hinzugefügt|
|`sensitive_data/personal_data/protection_concept/rights_management`|neu hinzugefügt|
|`sensitive_data/personal_data/protection_concept/workflow         `|neu hinzugefügt|

Zack-83 commented:
@torsten-rathmann bitte [Anleitung](https://github.com/rdmorganiser/rdmo-catalog/blob/master-rdmo2.x/shared/fodako/Dokumentation5.1.pdf) korrigieren:

* `sensitive_data/personal_data/protection_concept/rights_management` in der Liste der "neuen" Attribute einschließen
* `project/dataset/documentation`, `project/dataset/documentation/where`, `project/dataset/documentation_on_request`, `project/dataset/ipr/status`, `project/dataset/reuse_existing`, `project/legal_aspects/official_approval/statutatory_approval/conditions`, `project/costs/other_research_output`, `project/costs/other_research_output/personnel`, `project/costs/other_research_output/non_personnel`, `project/other_research_output`, `project/other_research_output/plan` aus der Liste streichen (in der Domäne doch schon vorhanden)

Zack-83 commented:
@yvgrossmann Die Präfixänderungen sind unproblematisch, oder?

Zack-83 commented:
@jmfrenzel Stimmst du zu, dass diese Attribute würdig sind?

jmfrenzel reviewed (changes requested):
@Zack-83 this PR suggests to add new pieces  of shared content to the community maintained resources.   I recommend to store this new content  in the shared folder.

jmfrenzel commented:
I'm ok with the cosmetic changes. Please submit them in a separate PR.

Zack-83 commented:
> @Zack-83 this PR suggests to add new pieces of shared content to the community maintained resources. I recommend to store this new content in the shared folder.

I agree that this is the general rule. The problem is that this content has the same prefix `https://rdmorganiser.github.io/terms` as the core RDMO content.  
It was a mistake from the beginning, but it has been there for years and it is used by FoDaKo, a quite big user group. How do you suggest solving this knot?

torsten-rathmann commented:
Thanks for extending the attribute list and for the hint that sensitive_data/personal_data/protection_concept/rights_management is missing in the documentation! 

Unfortunately I couldn't test the new attributes so far but in the file attributes.xml I found  missprint: "project/dataset/sensitive_data/personal_data/consent/how_obrained" instaead of "project/dataset/sensitive_data/personal_data/consent/how_obtained". Could we have a correction?

Zack-83 commented on rdmorganiser/domain/attributes.xml:
```diff
@@ -994,13 +1015,27 @@
 		<dc:comment/>
 		<parent dc:uri="https://rdmorganiser.github.io/terms/domain/project/dataset/sensitive_data/personal_data/consent"/>
 	</attribute>
+	<attribute dc:uri="https://rdmorganiser.github.io/terms/domain/project/dataset/sensitive_data/personal_data/consent/how_obrained">
+		<uri_prefix>https://rdmorganiser.github.io/terms</uri_prefix>
+		<key>how_obrained</key>
+		<path>project/dataset/sensitive_data/personal_data/consent/how_obrained</path>
+		<dc:comment>Added for the FoDaKo DFG DMP templates, version 5.1</dc:comment>
+		<parent dc:uri="https://rdmorganiser.github.io/terms/domain/project/dataset/sensitive_data/personal_data/consent"/>
+	</attribute>
```
```suggestion
	<attribute dc:uri="https://rdmorganiser.github.io/terms/domain/project/dataset/sensitive_data/personal_data/consent/how_obtained">
		<uri_prefix>https://rdmorganiser.github.io/terms</uri_prefix>
		<key>how_obtained</key>
		<path>project/dataset/sensitive_data/personal_data/consent/how_obtained</path>
		<dc:comment>Added for the FoDaKo DFG DMP templates, version 5.1</dc:comment>
		<parent dc:uri="https://rdmorganiser.github.io/terms/domain/project/dataset/sensitive_data/personal_data/consent"/>
	</attribute>
```

Zack-83 commented:
> I'm ok with the cosmetic changes. Please submit them in a separate PR.

Done https://github.com/rdmorganiser/rdmo-catalog/commit/942651a088e73b4377b9990b33bfeb0d1c3cf340

Zack-83 commented:
> Thanks for extending the attribute list and for the hint that sensitive_data/personal_data/protection_concept/rights_management is missing in the documentation!
> 
> Unfortunately I couldn't test the new attributes so far but in the file attributes.xml I found the following missprint: "project/dataset/sensitive_data/personal_data/consent/how_obrained" instaead of "project/dataset/sensitive_data/personal_data/consent/how_obtained". Could we have a correction?

I have corrected the missprint. As requested by @jmfrenzel I moved the 9 new attributes to a new file: https://github.com/rdmorganiser/rdmo-catalog/blob/FoDaKo-Attribute/shared/fodako/attributes-FoDaKo-dfg.xml Please update the documentation accordingly.

torsten-rathmann commented:
I have tested the new attribute file [shared/fodako/attributes-FoDaKo-dfg.xml](https://github.com/rdmorganiser/rdmo-catalog/blob/FoDaKo-Attribute/shared/fodako/attributes-FoDaKo-dfg.xml) from scratch and didn't find any issues. Thanks for the good work!
